### ISSUE TITLE
yum and apt repos are behind

### ISSUE THREAD

**Issue by Tyriar** [Microsoft/vscode#45868]
Currently 1.21.1 isn't published to the repos, I'm working with the repo team to resolve the problem. The 1.21.0 deb is also having a data mismatch, breaking the apt repo.

In the meantime you can install the rpm or deb manually by [downloading it from the website](https://code.visualstudio.com/download), this should always be the latest version.

---

Problems:

- [x] Fix 1.21.0 deb
- [x] Publish 1.21.1 deb
- [x] Publish 1.21.1 rpm

**Comment by toxinu** [comment]
Where are `deb` and `rpm` files on website?
I cannot find it.

**Comment by Tyriar** [comment]
@toxinu https://code.visualstudio.com/download

**Comment by el-duderino5** [comment]
Good day. Whenever you fix the problem, is there something we have to fix on our part? I am using 
Debian and, when I installed vscode using your .deb package, it also installed a vscode.list file in the /etc/apt/sources.list.d directory that points to your repo. Are we going to need to modify this file or is there something else we need to do after you fix the problem?

**Comment by Katana-Steel** [comment]
for the apt repo InRelease & Release.gpg wasn't updated:
(url: https://packages.microsoft.com/repos/vscode/dists/stable/)
```
../
main/                                              15-Mar-2018 18:02                   -
InRelease                                          14-Mar-2018 06:39                2802
Release                                            15-Mar-2018 18:02                2298
Release.gpg                                        14-Mar-2018 06:39                 455
```
resulting in this when running apt update:
```
E: Failed to fetch https://packages.microsoft.com/repos/vscode/dists/stable/main/binary-amd64/Packages.bz2  File has unexpected size (43829 != 43234). Mirror sync in progress? [IP: 13.91.48.226 443]
   Hashes of expected file:
    - Filesize:43234 [weak]
    - SHA512:1bd713bc0ff5f583b7427cd7f70bb378c324980df8db6a6480a03eade92b9a30ccdc4a0e58f35206035c073c6a64d3ca3775442d483a392c08a2918f85ee90f9
    - SHA256:e311c67fbc37c451043b20183bb3782b7a679d2fb488d5888f581b4393535934
    - SHA1:d86fe0862f1d85589ef2a620b4115951c028def0 [weak]
    - MD5Sum:755a820b77a2c96e347e9239f349abee [weak]
   Release file created at: Wed, 14 Mar 2018 06:38:24 +0000
```

**Comment by Tyriar** [comment]
Both 1.21.0 and 1.21.1 debs should now work. We believe the hash mismatch was caused by the repo partially being updated due to the use of a new API not validating the version correctly (the -epoch part of the version was being ignored).

**Comment by el-duderino5** [comment]
Hello @Tyriar . What I ended up doing was re-installing code by downloading the latest .deb from your page. I am on the latest version right now but, is there something else we, the users, need to modify on our machines so that whenever a new version of vscode comes out, we do not need to re-install again, starting by downloading the .deb and installing manually? Or can it finally can be upgraded in the future using the apt utilities? Was this issue only on your side?

Also, can you please comment on why `sudo apt-get install -f` doesn't work on my Debian system and can you please confirm if your setup documentation is outdated? If vscode needs dependencies, I am afraid that command won't cut it.

**Comment by joshsleeper** [comment]
@el-duderino5 
> ... is there something else we, the users, need to modify on our machines so that whenever a new version of vscode comes out, we do not need to re-install again ...

i believe the answer is "no". 
once the user has installed the appropriate package for their system (`deb` or `rpm`) then successive version of vs code should be available via both `apt-get` and `yum`/`dnf` via the repo added when the package was installed.

this has been the case for some time now (i use both platforms regularly) and this issue was only for tracking the fact that these repos have started falling behind a bit. there's usually a multi-day wait for the `rpm` release to show up in the repos (which may not be fixed anytime soon), and according to @Tyriar's comment just above...

> Both 1.21.0 and 1.21.1 debs should now work. We believe the hash mismatch was caused by the repo partially being updated due to the use of a new API not validating the version correctly (the -epoch part of the version was being ignored).
> -@Tyriar

... it sounds like there was just a hiccup with the `apt` repo that should now be resolved.

> Also, can you please comment on why sudo apt-get install -f doesn't work on my Debian system and can you please confirm if your setup documentation is outdated? If vscode needs dependencies, I am afraid that command won't cut it.

You should probably file a different issue for this since it has no bearing on this one.

**Comment by rmunn** [comment]
I can confirm that the 1.21.1 deb downloads and installs just fine now. I have a machine where I hadn't run updates in two weeks, so that machine missed the 1.21.0 update, and that machine installed 1.21.1 after a perfectly normal `sudo apt update; sudo apt upgrade` sequence.

**Comment by Tyriar** [comment]
The 1.21.1 rpm is now published, the issue was the repo admins recommended moving to an API as it was more stable. Turns out that was not true. There was also the added delay of manual signing. I went over to visit the team and emphasized the importance of getting rpm signing automated and improving stability. Hopefully they will work on it soon.

---

@el-duderino5 as @joshsleeper says, this was just a hiccup on the apt side, the deb typically goes out immediately with no problems. rpm is a different story though and there is pretty much always a delay, I added a note to the Linux setup docs about this as while I'm hopeful that it will improve soon, realistically it will be like this for some time https://github.com/Microsoft/vscode-docs/commit/edd77942d0a43d734ffed7aa51b0c13b98bc6744
